api/services/userService.js

```diff
@@ -21,6 +22,33 @@ const signUp = async (email, password, name) => {
   return user;
 };
 
+const signIn = async (email, password) => {
+  try {
+    const userInfo = await userDao.getUserByEmail(email);
+    const hashedPassword = await userInfo.password;
+
+    if (!hashedPassword) {
+      const error = new Error("EMAIL_NOT_FOUND");
+      error.statusCode = 404;
+      throw error;
+    }
+    const checkHash = bcrypt.compare(password, hashedPassword);
+    if (!checkHash) {
+      const error = new Error("WRONG_PASSWORD");
+      error.statusCode = 400;
+      throw error;
+    }
+    const userId = await userInfo.id;
+    const token = jwt.sign({ id: userId }, process.env.SECRET_KEY);
+
+    return token;
+  } catch (error) {
+    console.log(error);
+    throw new Error("FAILED_TO_LOGIN");
+  }
+};
```

Comment:
1. controller 계층에서 catchAsync 함수를 사용하고 있기 때문에, 해당 모듈인 service는 물론 dao에서도 try catch는 기본적으로 필요하지 않습니다.
2. userInfo 변수를 보면 `userDao.getUserByEmail(email);` 에서 return 되는 결과값이 배열을 제거한 이메일에 특정되는 오브젝트 `1개`가 반환될 것이므로 30번 라인의 `!hashedPassword` 보다 `!userInfo`로 작성하는게 더 좋을 것 같습니다.
3. await 가 있어야 할 곳과 없어도 되는 곳이 섞여 있는 것 같습니다. 아래 suggestion 확인 부탁드릴게요~

```suggestion
const signIn = async (email, password) => {
    const userInfo = await userDao.getUserByEmail(email);
    const hashedPassword = userInfo.password; // await 가 필요하지 않음

    if (!userInfo) { // 2번 피드백
      const error = new Error("EMAIL_NOT_FOUND");
      error.statusCode = 404;
      throw error;
    }
    
    const checkHash = await bcrypt.compare(password, hashedPassword); // await 필요
    
    if (!checkHash) {
      const error = new Error("WRONG_PASSWORD");
      error.statusCode = 400;
      throw error;
    }
    
    const userId = userInfo.id; // await 필요하지 않음
    const token = await jwt.sign({ id: userId }, process.env.SECRET_KEY); // await 필요

    return token;
};
```